### PR TITLE
Support for an extensible raw-value

### LINKED ISSUE #16: raw-value has been redefined as extensible
Another spec change that needs to be reflected in the implementation:
* https://github.com/ietf-rats/ietf-corim-cddl/pull/125

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #16

### CODE REVIEW

File: comid/rawvalue.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	if dm.Unmarshal(data, &rawValue) == nil {
+		o.val = rawValue
+		return nil
+	}
+
+	return fmt.Errorf("unknown raw-value (CBOR: %x)", data)
+}
+
+// UnmarshalJSON deserializes the type'n'value JSON object into the target RawValue.
+// The only supported type is "bytes" with value
```
Comment (thomas-fossati) [inline comid/rawvalue.go:44]:
```suggestion
// The only supported type is "bytes", corresponding to CBOR tag 560
```

File: comid/rawvalue.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package comid
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type RawValue struct {
```
Comment (yogeshbdeshpande) [inline comid/rawvalue.go:11]: Perhaps we should mention what this RawValue is for and its relevance in CoMID?

File: comid/rawvalue.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	val interface{}
+}
+
+type TaggedRawValueBytes []byte
```
Comment (yogeshbdeshpande) [inline comid/rawvalue.go:15]: Understand that the TaggedRaw Value is also Byte stream, but could you please also add what tag sits at the srtart?

File: comid/measurement.go [hunk 4]
```diff
@@ -97,7 +97,7 @@ type Mval struct {
 	SVN          *SVN      `cbor:"1,keyasint,omitempty" json:"svn,omitempty"`
 	Digests      *Digests  `cbor:"2,keyasint,omitempty" json:"digests,omitempty"`
 	OpFlags      *OpFlags  `cbor:"3,keyasint,omitempty" json:"op-flags,omitempty"`
-	RawValue     *[]byte   `cbor:"4,keyasint,omitempty" json:"raw-value,omitempty"`
+	RawValue     *RawValue `cbor:"4,keyasint,omitempty" json:"raw-value,omitempty"`
```
Comment (yogeshbdeshpande) [inline comid/measurement.go:100]: Wondering we set this incorrectly. I am little confused, as to why RawValue and RawValueMask is not logically clubbed into a single entity, aligning it with CDDL Schema.

I think, they are logically seen as grouped entity?

Comment (thomas-fossati) [reply comid/measurement.go:100]: saying
```
raw-value-group = (
  comid.raw-value => $raw-value-type-choice
  ? comid.raw-value-mask => raw-value-mask-type
 )
```
 and then
```
measurement-values-map = non-empty<{
  [...]
  ? raw-value-group
  [...]
}>
```
is just a different way to say
```
measurement-values-map = non-empty<{
  [...]
  ? ( comid.raw-value => $raw-value-type-choice, ? comid.raw-value-mask => raw-value-mask-type )
  [...]
}>
```

File: comid/rawvalue.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package comid
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
```
Comment (thomas-fossati) [inline comid/rawvalue.go:10]:
```suggestion

// RawValue models a $raw-value-type-choice.  For now, the only available type is bytes.
```

File: comid/rawvalue.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+type RawValue struct {
+	val interface{}
+}
+
```
Comment (thomas-fossati) [inline comid/rawvalue.go:14]:
```suggestion

// TaggedRawValueBytes is an alias for []byte to allow its automatic tagging
```

File: comid/rawvalue.go (hunk 2 above)
Comment (thomas-fossati) [reply comid/rawvalue.go:12]: sure, done in e1e9225

File: comid/rawvalue.go (hunk 3 above)
Comment (thomas-fossati) [reply comid/rawvalue.go:17]: done in e1e9225

**Comment by yogeshbdeshpande** [comment]
Please also raise github issue  for pending items on line 132 and 146 (for measurement.go)

**Review by yogeshbdeshpande — APPROVED** [review]

**Pull request merged**